keycloak/keycloak-admin-ui
# User fed wizard

## Motivation
Adds the wizard screens (just in storybook for now) for user federation ldap and kerberos.
LDAP wizard: https://marvelapp.com/prototype/ecjc94b/screen/72071048
Kerberos wizard: https://marvelapp.com/prototype/ecjc94b/screen/72098365

## Brief Description
Adds two stories for the LDAP and Kerberos wizards. Also adds a story for a wizard section header component.
The two user federation settings pages also had to be refactored because the wizard has a section description that doesn't show up on the settings pages. So each section was changed to use the new section header and to use a prop on whether to display the title and description.
Note that these wizards are designed as in-page wizards, so the next step will be to embed them into the content area of a "new user federation" page.

## Verification Steps
In Storybook, check the User Federation LDAP Wizard and User Federation Kerberos Wizard against the mockups. You can advance through the steps, but finishing the wizard is not implemented here.
In Storybook, there is now a Wizard Section Header. Check that the props correctly show/hide the specified text.
Under user federation, check each of the settings pages for LDAP and Kerberos are unchanged in their appearance.

## Checklist:

- [ x] Code has been tested locally by PR requester
- [ x] User-visible strings are using the react-i18next framework (useTranslation)
- [ x] Help has been implemented
- [ X] axe report has been run and resulting a11y issues have been resolved
- [na] Unit tests have been created/updated
- [ x] Formatting has been performed via prettier/eslint
- [ x] Type checking has been performed via 'yarn check-types'

## Additional Notes
Kerberos wizard: 
![kerberos-wizard](https://user-images.githubusercontent.com/19825616/103670163-38412d80-4f47-11eb-836a-56bedb8ab10e.gif)

LDAP wizard: 
![ldap-wizard](https://user-images.githubusercontent.com/19825616/103670180-3d05e180-4f47-11eb-85d9-fa23ba70254a.gif)
